dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestCriteria.cpp

```diff
@@ -2436,4 +2458,103 @@ TEST(DataInterfaceIIDMTest, testGeneratorCriteriaDataIIDMSum) {
+  };
+
+  std::vector<VLTestConfig> vlTestConfigArray = {
+                                  {criteria::CriteriaParams::LOCAL_VALUE, criteria::CriteriaParams::DYNAMIC, false, 2},
+                                  {criteria::CriteriaParams::LOCAL_VALUE, criteria::CriteriaParams::FINAL, true, 2},
+                                  {criteria::CriteriaParams::SUM, criteria::CriteriaParams::DYNAMIC, false, 1},
+                                  {criteria::CriteriaParams::SUM, criteria::CriteriaParams::FINAL, true, 1}
+                                };
+
+  for (VLTestConfig& vlTestConfig : vlTestConfigArray) {
+    boost::shared_ptr<criteria::CriteriaParams> criteriaParamsWithoutVoltageLevel = CriteriaParamsFactory::newCriteriaParams();
+    criteriaParamsWithoutVoltageLevel->setType(vlTestConfig.vlTestType);
+    criteriaParamsWithoutVoltageLevel->setScope(vlTestConfig.vlTestScope);
+    criteriaParamsWithoutVoltageLevel->setPMax(40);
+
+    boost::shared_ptr<criteria::CriteriaParams> criteriaParamsWithVoltageLevel = CriteriaParamsFactory::newCriteriaParams();
+    criteriaParamsWithVoltageLevel->setType(vlTestConfig.vlTestType);
+    criteriaParamsWithVoltageLevel->setScope(vlTestConfig.vlTestScope);
+    criteriaParamsWithVoltageLevel->setPMax(40);
+    criteriaParamsWithVoltageLevel->addVoltageLevel(criteriaVoltageLevel1);
+    criteriaParamsWithVoltageLevel->addVoltageLevel(criteriaVoltageLevel2);
+
+    // Test on LoadCriteria
+    boost::shared_ptr<DataInterface> dataWithLoads = createDataItfFromNetworkCriteria(createBusBreakerNetworkWithLoads(190, 190, 100, 100));
+    exportStates(dataWithLoads);
+    std::vector< boost::shared_ptr<LoadInterface> > loads = dataWithLoads->getNetwork()->getVoltageLevels()[0]->getLoads();
+
+    LoadCriteria loadCriteriaWithoutVoltageLevel(criteriaParamsWithoutVoltageLevel);
+    for (size_t loadIdx1 = 0; loadIdx1 < loads.size(); ++loadIdx1)
+      loadCriteriaWithoutVoltageLevel.addLoad(loads[loadIdx1]);
+    loadCriteriaWithoutVoltageLevel.checkCriteria(0, vlTestConfig.isFinalStep);
+    std::vector<std::pair<double, std::string> > loadFailingCriteriaWithoutVL = loadCriteriaWithoutVoltageLevel.getFailingCriteria();
+    ASSERT_EQ(loadFailingCriteriaWithoutVL.size(), vlTestConfig.failingCriteriaSize);
+    checkNoLoadIsTestedSeveralTimes(loadFailingCriteriaWithoutVL);
+
+    LoadCriteria loadCriteriaWithVoltageLevel(criteriaParamsWithVoltageLevel);
+    for (size_t loadIdx2 = 0; loadIdx2 < loads.size(); ++loadIdx2)
+      loadCriteriaWithVoltageLevel.addLoad(loads[loadIdx2]);
+    loadCriteriaWithVoltageLevel.checkCriteria(0, vlTestConfig.isFinalStep);
+    std::vector<std::pair<double, std::string> > loadFailingCriteriaWithVL = loadCriteriaWithVoltageLevel.getFailingCriteria();
+    ASSERT_EQ(loadFailingCriteriaWithVL.size(), vlTestConfig.failingCriteriaSize);
+    checkNoLoadIsTestedSeveralTimes(loadFailingCriteriaWithVL);
+    ASSERT_EQ(loadFailingCriteriaWithoutVL, loadFailingCriteriaWithVL);
```

Comment:
This test is not testing that the number of loads without VL is bigger tahn with VL

Reply:
If this test returns OK, it means the criteria without VL didn't filter any loads since the number of criteria in loadFailingCriteriaWithoutVL and loadFailingCriteriaWithVL is the same.

Reply:
shouldn't the number of loads in loadFailingCriteriaWithVL be inferior?

Reply:
Inferior or equal

Reply:
yes but so that this tests make sure that we filter it should be inferior in this case

Reply:
I will try to be clearer: I would like a test where some loads are filtered by the VLs along to a test where the VL are not used and thus we test that all loads are tested

Reply:
I can implement another test so that filtered loads (when there are VLs) are fewer than not filtered VLs but it won't test the new functionality I implemented. It will only test that the filter works.

Reply:
I disagree: It will test that if there is a filter then your new code is not used